toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/ConfigPanel.java

```diff
@@ -2,408 +2,169 @@
-				textButtonPanelChrome.add(textFieldChrome);
-				textButtonPanelChrome.add(iconPanelChrome);
-				errorLabelChrome = errorLabel;
-				
-				chromePanel = panel;
-				chromePanel.add(textButtonPanelChrome);
-				chromePanel.add(errorLabelChrome);
-			}
-			if(strKey.contains(webAppName)) {
-				textFieldWebApp = textField;
-				iconPanelWebApp = iconPanel;
-				iconValidWebApp = iconValid;
-				iconNotValidWebApp = iconNotValid;
-				textButtonPanelWebApp = textButtonPanel;
-				textButtonPanelWebApp.add(textFieldWebApp);
-				textButtonPanelWebApp.add(iconPanelWebApp);
-				errorLabelWebApp = errorLabel;
-
-				webAppPanel = panel;
-				webAppPanel.add(textButtonPanelWebApp);
-				webAppPanel.add(errorLabelWebApp);
-			}
-			if(strKey.contains(recorderName)) {
-				textFieldRecorder = textField;
-				iconPanelRecorder = iconPanel;
-				iconValidRecorder = iconValid;
-				iconNotValidRecorder = iconNotValid;
-				textButtonPanelRecorder = textButtonPanel;
-				textButtonPanelRecorder.add(textFieldRecorder);
-				textButtonPanelRecorder.add(iconPanelRecorder);
-				errorLabelRecorder = errorLabel;
-
-				recorderPanel = panel;
-				recorderPanel.add(textButtonPanelRecorder);
-				recorderPanel.add(errorLabelRecorder);
-			}
-			if(strKey.contains(apiKeyName)) {
-				textFieldApiKey = textField;
-				iconPanelApiKey = iconPanel;
-				textButtonPanelApiKey = textButtonPanel;
-				textButtonPanelApiKey.add(textFieldApiKey);
-				textButtonPanelApiKey.add(iconPanelApiKey);
-				errorLabelApiKey = errorLabel;
-
-				apiKeyPanel = panel;
-				apiKeyPanel.add(textButtonPanelApiKey);
-				apiKeyPanel.add(errorLabelApiKey);
-			}
-		}
-		
-
-		JPanel buttonPanel = new JPanel();
-		buttonPanel.setAlignmentX(LEFT_ALIGNMENT);
-		buttonPanel.setLayout(new BoxLayout(buttonPanel, BoxLayout.LINE_AXIS));
-		buttonPanel.add(Box.createHorizontalGlue());
-		JButton okButton = new JButton("Ok");
-		
-		okButton.addActionListener(new ActionListener() {
-
-			public void actionPerformed(
-				ActionEvent e) {
-				saveAndClose();
-			}
-		});
-		JButton cancelButton = new JButton("Cancel");
-		cancelButton.addActionListener(new ActionListener() {
+		this.setVisible(true);
+	}
 
-			public void actionPerformed(
-				ActionEvent e) {
-				close();
-			}
-		});
+	private void buildContentPanel() throws IOException {
 		
-
-		JButton tryButton = new JButton("Test");
-		tryButton.addActionListener(new ActionListener() {
-			public void actionPerformed(ActionEvent arg0) {
-
-		        for(Object key : EnumerationUtils.toList(properties.propertyNames())) 
-				{
-					String strKey = (String) key;
-					try {
-						testIconValid(strKey, false);
-						secondPane.repaint();
-						secondPane.revalidate();
-					} catch (IOException e) {
-						LOG.error(e.getMessage(), e);
-					}
-				}		        
-		        NotificationManager.showMessage("The parameters have been tested !");
-
-			}
-			
-		});
+		this.boxFields = new HashMap<String, AbstractPanel>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)